src/_TextInputInnerAction.tsx

```diff
@@ -5,7 +5,7 @@ import {Button, IconButton, ButtonProps} from './Button'
 import Tooltip from './Tooltip'
 import {BetterSystemStyleObject, merge, SxProp} from './sx'
 
-type TextInputActionProps = Omit<React.HTMLProps<HTMLButtonElement>, 'aria-label' | 'size'> & {
+type TextInputActionProps = Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'aria-label' | 'size'> & {
```

Comment:
this needed to be narrowed to avoid the dom `as` property that `HTMLProps` defines as `as: string` from being improperly considered as a value that gets passed to `Button`.